### PR TITLE
fix: untranslatable strings in trade component and broken ui

### LINKED ISSUE #3083: Trade component errors not translatable and breaking the UI when too long
### Overview

The error message "[Sell/Buy] asset not supported by wallet" displayed inside the Trade component as a button, is not fully translatable.

This Button/Error also does not adapt to longer strings, which means it gets truncated and looks broken in certain languages as you can see in the screenshot.

image

This error is also briefly displayed each time the page is refreshed (tested with MetaMask and ShapeShift wallet). It shows up while the component/page is loading before any asset is displayed. That's why it caught my attention, and I would guess other users will notice/be alarmed by it too. I think it shouldn't be displayed until assets displayed (and only if needed of course).

### References and additional details

'Sell asset' and 'Buy asset' in the next portion of code should be made translatable like the rest of the underlying string used `trade.errors.assetNotSupportedByWallet` :

https://github.com/shapeshift/web/blob/b99fb8e1ca44c42d3a4620d04d3624bf2ac06e29/src/components/Trade/TradeInputV2.tsx#L296-L305

If possible adapt the button so it can accept longer strings, so translations won't cause it to break as seen in the screenshot. It currently has a CSS rule for `white-space: nowrap;` which could be changed to allow wrapping... obviously this requires more UI testing.

### Acceptance Criteria

* Both error messages are fully translatable and present **only in the English** `main.json` file.
Translations in other languages will be provided by the Globalization WS in one of the weekly PR for translations, after this issue is resolved in the base language.
* The button/error can be displayed within the bounds of the button box even for longer strings.
* Avoid showing the error while until the assets are loaded and it is actually an error that should be communicated to the user.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by firebomb1**
## Description

* Makes four of the errors of the Trade component fully translatable
* Adds support for multiple lines in Button component by adding new `size` option in the chakra-ui Button theme and use it for the Trade component button.
* Adds French translation for testing purposes for the long error messages in question.
* Displays a "Loading..." status in the Trade button instead of a temporary/meaningless error message, until we can display a meaningful one (or no error, if none happened during the loading phase).

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #3083 

## Risk

Low, the UI changes are done through a new `size` option for Button components which isn't used anywhere else than the Trade component.

### Operations

The error message that previously was partially untranslatable is displayed just when you refresh the app in the browser, right when the page loads and all assets aren't yet loaded (this part of #3083 is not fixed in this PR yet).

The steps to reproduce are to connect a wallet (tested with MetaMask and ShapeShift wallets), refresh the page of the app while on the Dashboard and look at the Trade section in the right column displaying the error message briefly. Select the French language (Français) in settings, and do the previous steps to witness a longer message fit in the button on multiple lines.

## Screenshots (if applicable)

Before the changes (note "Sell" which is untranslatable/untranslated and the text overflowing outside of the button):
image

After the changes:
image

### CODE REVIEW

File: src/components/Trade/TradeInput.tsx [hunk 1]
```diff
@@ -297,35 +299,54 @@ export const TradeInput = () => {
     const minLimit = `${bnOrZero(quote?.minimum).decimalPlaces(6)} ${quote?.sellAsset.symbol}`
 
     if (!wallet) return 'common.connectWallet'
+    if (
+      sellTradeAsset?.asset?.chainId === undefined ||
```
Comment (0xdef1cafe) [inline src/components/Trade/TradeInput.tsx:303]: we don't need to check for `.chainId` on the `asset` here - the `asset` is an object which is truthy.

you can just do

`if (!sellTradeAsset?.asset || !buyTradeAsset?.asset) ...`

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
just some small requested changes, nothing major, i get what you're doing here

**Comment by firebomb1** [comment]
Ok, thanks for the review!

**Review by 0xdef1cafe — APPROVED** [review]

File: src/components/Trade/TradeInput.tsx (hunk 1 above)
Comment (firebomb1) [reply src/components/Trade/TradeInput.tsx]: @0xdef1cafe Regarding this check, it seems that it introduced a small regression, for an asset that is not supported by the wallet the button keeps displaying "Loading..." as `sellTradeAsset.asset`/`buyTradeAsset.asset` never get set in this case apparently. For example you can try selecting Bitcoin as an asset using MetaMask. 

Would it be possible to rather check if the attempt to load the asset is finished, display "Loading..." until it's done, and if it's done start showing the potential error messages?

**Pull request merged**